File: Modules/Common/Techs/SIE.Web.Tech/Routings/Commands/AddRouting.cs
using SIE.Domain;
using SIE.Security;
using SIE.Tech.Routings;
using SIE.Web.Command;

namespace SIE.Web.Tech.Routings.Commands
{
    /// <summary>
    /// 添加工艺路线命令
    /// </summary>
    [JsCommand("SIE.Web.Tech.Routings.Commands.AddRouting")]
    [AllowAnonymous]
    public class AddRouting : ViewCommand
    {
        /// <summary>
        /// 添加工艺路线命令执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var routing = args.Data.ToJsonObject<Routing>();
            routing.PersistenceStatus = PersistenceStatus.New;
            RF.Save(routing);
            return new
            {
                Id = routing.Id,
                CategoryId = routing.CategoryId,
                Name = routing.Name,
                Description = routing.Description,
                DefaultVersionId = routing.DefaultVersionId,
                text = routing.Name,
                leaf = false,
            };
        }
    }
}

File: Modules/Common/Techs/SIE.Web.Tech/Routings/Commands/AddRouting.js
SIE.defineCommand('SIE.Web.Tech.Routings.Commands.AddRouting', {
    extend: 'SIE.cmd.Command',
    meta: { text: "", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    execute: function (view, source,win) { 
        var iptdata = {
            CategoryId: source.categoryId,
            Name: source.name,
            Description: source.desc
        };
        view.execute({
            command: Ext.getClassName(this),
            data: iptdata,
            success: function (res) {
                //在树中添加工艺路线节点
                var routing = res.Result;
                var categroyNode = source.Category;
                var node = categroyNode.appendChild(routing);
                var treePanel = categroyNode.getOwnerTree(); 
                treePanel.getSelectionModel().select(node);
                var treeview = treePanel.getView(); 
                categroyNode.expand();
                node.expand();
                treeview.focusRow(node); 
                treePanel.fireEvent('itemclick', treeview, node, treeview.getNodeByRecord(node), treeview.indexOfRow(node));
                if (win != null)
                    win.close();
            }
        });
    }
});